include/SDL3/SDL_main_impl.h

```diff
@@ -0,0 +1,202 @@
+/*
+  Simple DirectMedia Layer
+  Copyright (C) 1997-2022 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely, subject to the following restrictions:
+
+  1. The origin of this software must not be misrepresented; you must not
+     claim that you wrote the original software. If you use this software
+     in a product, an acknowledgment in the product documentation would be
+     appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+
+#ifndef SDL_main_windows_h_
+#define SDL_main_windows_h_
+
+#if !defined(SDL_main_h_)
+#error "This header should not be included directly, but only via SDL_main.h!"
+#endif
+
+/* if someone wants to include SDL_main.h but doesn't want the main handing magic,
+   (maybe to call SDL_RegisterApp()) they can #define SDL_MAIN_HANDLED first
+   SDL_MAIN_NOIMPL is for SDL-internal usage (only affects implementation,
+   not definition of SDL_MAIN_AVAILABLE etc in SDL_main.h) and if the user wants
+   to have the SDL_main implementation (from this header) in another source file
+   than their main() function, for example if SDL_main requires C++
+   and main() is implemented in plain C */
+#if !defined(SDL_MAIN_HANDLED) && !defined(SDL_MAIN_NOIMPL)
+
+/* the implementations below must be able to use the implement real main(), nothing renamed
+   (the user's main() will be renamed to SDL_main so it can be called from here) */
+#ifdef main
+#  undef main
+#endif /* main */
+
+#if defined(__WIN32__) || defined(__GDK__)
+
+/* these defines/typedefs are needed for the WinMain() definition */
+#ifndef WINAPI
+#define WINAPI __stdcall
+#endif
+
+#include <SDL3/begin_code.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct HINSTANCE__ * HINSTANCE;
+typedef char* LPSTR;
+
+#ifndef __GDK__ /* this is only needed for Win32 */
+
+#if defined(_MSC_VER)
+/* The VC++ compiler needs main/wmain defined */
+# define console_ansi_main main
```

Comment:
I'm still not sure if console_ansi_main (and console_wmain) shouldn't just *always* be called main and wmain, instead of just when compiling with MSVC.

Reply:
Sure, it's worth trying here.